Content.Server/GameObjects/Components/AdvertiseComponent.cs

```diff
@@ -0,0 +1,144 @@
+using Robust.Shared.Log;
+using Robust.Shared.Prototypes;
+using Robust.Shared.Random;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.ViewVariables;
+
+namespace Content.Server.GameObjects.Components
+{
+    [RegisterComponent]
+    public class AdvertiseComponent : Component
+    {
+        public override string Name => "Advertise";
+
+        private CancellationTokenSource _cancellationSource = new();
+
+        /// <summary>
+        /// Minimum time to wait before saying a new ad, in seconds. Has to be larger than or equal to 1.
+        /// </summary>
+        [ViewVariables]
+        [field: DataField("minWait")]
+        private int MinWait { get; } = 480; // 8 minutes
+
+        /// <summary>
+        /// Maximum time to wait before saying a new ad, in seconds. Has to be larger than or equal
+        /// to <see cref="MinWait"/>
+        /// </summary>
+        [ViewVariables]
+        [field: DataField("maxWait")]
+        private int MaxWait { get; } = 600; // 10 minutes
+
+        [field: DataField("pack")]
+        private string PackPrototypeId { get; } = string.Empty;
+
+        private List<string> _advertisements = new();
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            IoCManager.Resolve<IPrototypeManager>().TryIndex(PackPrototypeId, out AdvertisementsPackPrototype? packPrototype);
+
+            // Load advertisements pack
+            if (string.IsNullOrEmpty(PackPrototypeId) || packPrototype == null)
+            {
+                // If there is no pack, log a warning and don't start timer
+                Logger.Warning($"{Owner} has {Name}Component but no advertisments pack.");
+                return;
+            }
+
+            _advertisements = packPrototype.Advertisements;
+
+            // Do not start timer if advertisement list is empty
+            if (_advertisements.Count == 0)
+            {
+                // If no advertisements could be loaded, log a warning and don't start timer
+                Logger.Warning($"{Owner} tried to load advertisements pack without ads.");
+                return;
+            }
+
+            // Throw exception if MinWait is smaller than 1.
+            if (MinWait < 1)
+            {
+                throw new PrototypeLoadException($"{Owner} has illegal minWait for {Name}Component: {MinWait}.");
+            }
+
+            // Throw exception if MinWait larger than MaxWait.
+            if (MinWait > MaxWait)
+            {
+                throw new PrototypeLoadException($"{Owner} should have minWait greater than or equal to maxWait for {Name}Component.");
+            }
+
+            // Start timer at initialization, without MinWait bound
+            RefreshTimer(false);
+        }
+
+        public override void OnRemove()
+        {
+            _cancellationSource.Cancel();
+            _cancellationSource.Dispose();
+
+            base.OnRemove();
+        }
```

Comment:
Sorry I remembered one more thing that I made into a separate issue as I keep forgetting about it.

https://github.com/space-wizards/space-station-14/issues/3869

You should probably do these in Startup and Shutdown instead so if an admin can disable / enable advertising it's easy to do so.

Reply:
NVM turns out it's being deprecated